app/grandchallenge/components/views.py

```
@@ -472,3 +500,134 @@ def get(self, request, *args, **kwargs):
+        help_text = request.GET.get("help-text")
+        current_value = request.GET.get("current-value")
+
+        match widget_choice:
+            case FileWidgetChoices.FILE_SEARCH:
+                html_content = render_to_string(
+                    FileSearchWidget.template_name,
+                    {
+                        "widget": FileSearchWidget().get_context(
+                            name=prefixed_interface_slug,
+                            value=None,
+                            attrs={
+                                "help_text": help_text if help_text else None,
+                            },
+                        )["widget"],
+                    },
+                )
+                return HttpResponse(html_content)
+            case FileWidgetChoices.FILE_UPLOAD:
+                html_content = render_to_string(
+                    UserUploadSingleWidget.template_name,
+                    {
+                        "widget": UserUploadSingleWidget().get_context(
+                            name=prefixed_interface_slug,
+                            value=None,
+                            attrs={
+                                "id": interface,
+                                "help_text": _join_with_br(
+                                    help_text if help_text else None,
+                                    file_upload_text,
+                                ),
+                            },
+                        )["widget"],
+                    },
+                )
+                return HttpResponse(html_content)
+            case FileWidgetChoices.FILE_SELECTED:
+                try:
+                    qs = UserUpload.objects.filter(pk=current_value)
+                except ValidationError as e:
+                    if "uuid" not in e.message.lower():
+                        raise
+                    qs = ComponentInterfaceValue.objects.filter(
+                        pk=current_value
+                    )

```

See my other comment. Why not try to convert `current_value` to a uuid and choose the model according to whether that succeeds?

---

Currently checking for integer type first. When it's not, I'm assuming it must be a uuid and check if a user upload object exists. I could do it the other way around with:
```python
try:
    UUID(current_value)
except ValueError:
    # assume it's an integer type pk
    # check if civ object exists
else:
    # check if user upload object exists
```

---

Doesn't matter which way around you do it I guess. We know that the value should either be a user upload pk or a civ pk, so maybe it's worth checking for both explicitly, and else log an error to sentry (without raising the error for the user, the decompressed value will just be [None, None] in that case). What do you think? 

To log the error to sentry without raising it, you could do `logger.error(e, exc_info=True)`.

---

That's great! I did not realize we could log errors like that
I tried to incorporate it. Could you check if I did it right?

---

I take it back. I was under the wrong impression that this was about the decompress() function on the widget and not the view. In the view, we should raise the error or else the user doesn't get feedback that something went wrong and might keep trying and trying, DOSing the site.